basic_bootloader/src/bootloader/mod.rs

```diff
@@ -476,4 +487,77 @@ impl<S: EthereumLikeTypes> BasicBootloader<S> {
             }
         }
     }
+
+    #[cfg(feature = "interop")]
+    fn process_interop_roots(
+        system: &mut System<S>,
+        system_functions: &mut HooksStorage<S, S::Allocator>,
+        memories: &mut RunnerMemoryBuffers,
+        tracer: &mut impl Tracer<S>,
+    ) -> Bytes32
+    where
+        S::IO: IOSubsystemExt,
+    {
+        use system_hooks::addresses_constants::L2_INTEROP_ROOT_STORAGE_ADDRESS;
+        let mut interop_root_hasher = crypto::sha3::Keccak256::new();
+
+        // Block of code needed for interop.
+        // We need to add interop roots to the interop root storage.
+        // We do it by calling the addInteropRoot function.
+        // The function is defined in the InteropRootStorage contract.
+        // The function is called with the chainId, blockOrBatchNumber, and the sides.
+        // The sides are the interop roots.
+        // The chainId is the chainId of the interop root.
+        // The blockOrBatchNumber is the block number of the interop root.
+        //
+        // We also compute the rolling hash of the interop roots and include it as part of the public input
+
+        system
+            .get_interop_roots()
+            .iter()
+            .filter(|interop_root| interop_root.chain_id != 0 && interop_root.block_number != 0)
+            .for_each(|interop_root| {
+                let mut data = [0u8; 160];
+                data[24..32].copy_from_slice(&interop_root.chain_id.to_be_bytes());
+                data[56..64].copy_from_slice(&interop_root.block_number.to_be_bytes());
+                data[92..96].copy_from_slice(&32u32.to_be_bytes());
+                data[124..128].copy_from_slice(&1u32.to_be_bytes());
+                data[128..160].copy_from_slice(&interop_root.root[0].as_u8_ref());
+                interop_root_hasher.update(&data);
+
+                data[92..96].copy_from_slice(&96u32.to_be_bytes());
+
+                // fb6200c6: function addInteropRoot(uint256 chainId, uint256 blockOrBatchNumber, bytes32[] calldata sides) external;
+                let mut calldata = [0xfb, 0x62, 0x00, 0xc6].to_vec();
```

Comment:
maybe you can put selector just at the beginning of data - so you can avoid memcopy